api/wasm/cpp/proxy_wasm_externs.h

```diff
@@ -77,6 +77,22 @@ extern "C" uint32_t proxy_getMetadataStruct(MetadataType type, const char* name_
 extern "C" uint32_t proxy_setMetadataStruct(MetadataType type, const char* name_ptr, size_t name_size,
                                             const char* value_ptr, size_t value_size);
 
+// Generic selectors
+extern "C" uint32_t proxy_resolveSelector0(const char* arg0_ptr, size_t arg0_size,
+                                           const char** value_ptr_ptr, size_t* value_size_ptr);
+extern "C" uint32_t proxy_resolveSelector1(const char* arg0_ptr, size_t arg0_size,
+                                           const char* arg1_ptr, size_t arg1_size,
+                                           const char** value_ptr_ptr, size_t* value_size_ptr);
+extern "C" uint32_t proxy_resolveSelector2(const char* arg0_ptr, size_t arg0_size,
+                                           const char* arg1_ptr, size_t arg1_size,
+                                           const char* arg2_ptr, size_t arg2_size,
+                                           const char** value_ptr_ptr, size_t* value_size_ptr);
+extern "C" uint32_t proxy_resolveSelector3(const char* arg0_ptr, size_t arg0_size,
```

Comment:
This is returning a uint32_t but the API functions are ignoring the return value.  What is it?  We should return some error code.   But with a name like "resolve" I would expect some sort of cursor.   Instead of resolving this seems to be returning a value.  Shouldn't it be called "get" then?

Reply:
I think absence of a value will become important, so we do need an error code for the resolution. It gets lost somewhere since it makes C++ interface ugly. Let me see if I can make it better.